rust/main/lander/src/adapter/chains/radix/adapter.rs

```diff
@@ -1,21 +1,46 @@
 #[cfg(test)]
 pub mod tests;
 
-use std::{sync::Arc, time::Duration};
+use std::{str::FromStr, sync::Arc, time::Duration};
 
-use scrypto::network::NetworkDefinition;
+use core_api_client::models::FeeSummary;
+use ethers::utils::hex;
+use gateway_api_client::models::{CompiledPreviewTransaction, TransactionPreviewV2Request};
+use radix_transactions::{
+    model::{IntentHeaderV2, TransactionHeaderV2, TransactionPayload},
+    prelude::{
+        DetailedNotarizedTransactionV2, ManifestBuilder, TransactionBuilder, TransactionV2Builder,
+    },
+    signing::PrivateKey,
+};
+use scrypto::{
+    address::AddressBech32Decoder,
+    crypto::IsHash,
+    math::{CheckedMul, Decimal, SaturatingAdd},
+    network::NetworkDefinition,
+    prelude::{
+        manifest_decode, ManifestArgs, ManifestCustomValue, ManifestCustomValueKind, ManifestValue,
+    },
+    types::{ComponentAddress, Epoch},
+};
 use uuid::Uuid;
 
-use hyperlane_core::H512;
-use hyperlane_radix::{RadixProviderForLander, RadixSigner, RadixTxCalldata};
+use hyperlane_core::{ChainCommunicationError, ChainResult, H256, H512};
+use hyperlane_radix::{RadixProvider, RadixProviderForLander, RadixSigner, RadixTxCalldata};
 
 use crate::{
-    adapter::{AdaptsChain, GasLimit, RadixTxPrecursor, TxBuildingResult},
+    adapter::{
+        chains::radix::{Precursor, VisibleComponents},
+        AdaptsChain, GasLimit, RadixTxPrecursor, TxBuildingResult,
+    },
     payload::PayloadDetails,
     transaction::{Transaction, TransactionUuid, VmSpecificTxData},
     DispatcherMetrics, FullPayload, LanderError, TransactionDropReason, TransactionStatus,
 };
 
+// the number of simulate calls we do to get the necessary addresses
+const NODE_DEPTH: usize = 5;
```

Comment:
Does it works for any contract, apart from Hyperlane Mailbox?

Reply:
How does the method `simulate_tx` depend on Mailbox? It seems Mailbox-independent at the glance.